bin/ch/WScriptJsrt.cpp

```diff
@@ -1371,6 +1372,48 @@ JsValueRef __stdcall WScriptJsrt::SleepCallback(JsValueRef callee, bool isConstr
     return returnValue;
 }
 
+JsValueRef __stdcall WScriptJsrt::GetProxyPropertiesCallback(JsValueRef callee, bool isConstructCall, JsValueRef *arguments, unsigned short argumentCount, void *callbackState)
+{
+    HRESULT hr = E_FAIL;
+    JsValueRef returnValue = JS_INVALID_REFERENCE;
+    JsErrorCode errorCode = JsNoError;
+
+    IfJsrtErrorSetGo(ChakraRTInterface::JsGetUndefinedValue(&returnValue));
+
+    if (argumentCount > 1)
+    {
+        bool isProxy = false;
+        JsValueRef target;
+        JsValueRef handler;
+        ChakraRTInterface::JsGetProxyProperties(arguments[1], &isProxy, &target, &handler);
+
+        if (isProxy)
+        {
+            JsPropertyIdRef targetProperty;
+            JsPropertyIdRef handlerProperty;
+            
+            IfJsrtErrorSetGo(CreatePropertyIdFromString("target", &targetProperty));
+            IfJsrtErrorSetGo(CreatePropertyIdFromString("handler", &handlerProperty));
+            IfJsrtErrorSetGo(ChakraRTInterface::JsCreateObject(&returnValue));
+
+            if (target == nullptr)
+            {
+                JsValueRef undefined = JS_INVALID_REFERENCE;
+                IfJsrtErrorSetGo(ChakraRTInterface::JsGetUndefinedValue(&undefined));
+                IfJsrtErrorSetGo(ChakraRTInterface::JsSetProperty(returnValue, targetProperty, undefined, true));
+                IfJsrtErrorSetGo(ChakraRTInterface::JsSetProperty(returnValue, handlerProperty, undefined, true));
+            }
+            else
+            {
+                IfJsrtErrorSetGo(ChakraRTInterface::JsSetProperty(returnValue, targetProperty, target, true));
+                IfJsrtErrorSetGo(ChakraRTInterface::JsSetProperty(returnValue, handlerProperty, handler, true));
```

Comment:
It's a little bit weird to me to set both of these properties in both sides of the conditional. What about setting `target` and `handler` to `undefined` if they're `JS_INVALID_REFERENCE` and having one set of `JsSetProperty` which always gets called?